gpt_researcher/utils/tools.py

```
@@ -0,0 +1,275 @@
+                tool_result = "Tool execution failed"
+                for tool in tools:
+                    if tool.name == tool_name:
+                        try:
+                            if hasattr(tool, 'ainvoke'):
+                                tool_result = await tool.ainvoke(tool_args)
+                            elif hasattr(tool, 'invoke'):
+                                tool_result = tool.invoke(tool_args)
+                            else:
+                                tool_result = await tool(**tool_args) if asyncio.iscoroutinefunction(tool) else tool(**tool_args)
+                            break
+                        except Exception as e:
```

Ruff (0.14.3)

Do not catch blind exception: `Exception`

(BLE001)